runtime/src/message_processor.rs

```diff
@@ -560,6 +589,7 @@ impl MessageProcessor {
             self.programs.clone(), // get rid of clone
             log_collector,
             self.is_cross_program_supported,
+            COMPUTE_BUDGET,
```

Comment:
I'm a little worried about how we'll deal with tweaking COMPUTE_BUDGET and MAX_INVOCATION_DEPTH in the future.  These can cause diverging behavior during a rolling upgrade and changes should be triggered on some slot/epoch boundary.  It'd be nice to consider how that will play out now, since technically right now COMPUTE_BUDGET is u64::MAX, and reducing might break existing programs so also needs to be  triggered on some slot/epoch boundary

Reply:
Agreed, we need a way to adjust these in the future.  These changes mostly mimic the current budget which isn't `u64::MAX` but instead `100_000` BPF instructions.  The only difference for existing programs is a small cost for the logging syscalls.  Programs using CPI are the ones that will see a bigger change (cost added for `create_progrm_address`, `invoke`, and charged for the cost programs they call incur.  I was planning to put in the rolling update stuff later when we need it and was focused on getting these CPI caps in place first.